src/components/AvatarWithImagePicker.js

```diff
@@ -116,166 +116,118 @@ const defaultProps = {
     originalFileName: '',
 };
 
-class AvatarWithImagePicker extends React.Component {
-    constructor(props) {
-        super(props);
-        this.animation = new SpinningIndicatorAnimation();
-        this.setError = this.setError.bind(this);
-        this.isValidSize = this.isValidSize.bind(this);
-        this.showAvatarCropModal = this.showAvatarCropModal.bind(this);
-        this.hideAvatarCropModal = this.hideAvatarCropModal.bind(this);
-        this.state = {
-            isMenuVisible: false,
-            validationError: null,
-            phraseParam: {},
-            isAvatarCropModalOpen: false,
-            imageName: '',
-            imageUri: '',
-            imageType: '',
+function AvatarWithImagePicker(props) {
+    const animation = new SpinningIndicatorAnimation();
+    const [isMenuVisible, setIsMenuVisible] = useState(false);
+    const [validationError, setValidationError] = useState(null);
+    const [phraseParam, setPhraseParam] = useState({});
+    const [isAvatarCropModalOpen, setIsAvatarCropModalOpen] = useState(false);
+    const [imageName, setImageName] = useState('');
+    const [imageUri, setImageUri] = useState('');
+    const [imageType, setImageType] = useState('');
+    const anchorRef = useRef();
+
+    useEffect(() => {
+        if (props.isUploading) {
+            animation.start();
+        }
+        
+        return () => {
+            animation.stop();
         };
-        this.anchorRef = React.createRef();
-    }
+    }, [props.isUploading]);
 
-    componentDidMount() {
-        if (!this.props.isUploading) {
-            return;
+    useEffect(() => {
+        if (!props.isFocused) {
+            setError(null, {});
         }
+    }, [props.isFocused]);
 
-        this.animation.start();
-    }
+    const setError = (error, phraseParam) => {
```

Comment:
```suggestion
    /**
     * @param {String} error
     * @param {Object} phraseParam
     */
    const setError = (error, phraseParam) => {
```

let's keep the comments